guides/source/active_storage_overview.md

```diff
@@ -662,6 +662,35 @@ are stored before the form is submitted, they can be used to retain uploads when
 <%= form.file_field :avatar, direct_upload: true %>
 ```
 
+## Querying
+
+Active Storage attachments are just ordinary Active Record associations behind the scenes, so you can use the usual `joins` / `where` toolbox to look up records whose attachments meet specific criteria.
+
+### `has_one_attached`
+
+`has_one_attached` creates a `has_one` association called `"<name>_attachment"` and a `has_one :through` association called `"<name>_blob"`.
+Need every user whose avatar is a PNG?
+
+```ruby
+User
+  .joins(:avatar_blob)
+  .where(active_storage_blobs: { content_type: "image/png" })
```

Comment:
I strongly dislike writing Ruby like this and don't want to encourage it in our docs.

```suggestion
User.joins(:avatar_blob).where(active_storage_blobs: { content_type: "image/png" })
```

Reply:
Fair!